networkx/algorithms/isomorphism/vf2pp.py

```diff
@@ -189,7 +221,14 @@ def vf2pp_all_isomorphisms(G1, G2, node_labels=None, default_label=None):
 
     # Initialize the stack
     stack = []
-    candidates = iter(_find_candidates(node_order[0], graph_params, state_params))
+    if not G1.is_directed():
+        candidates = iter(
+            _find_candidates(node_order[0], graph_params, state_params, G1_degree)
+        )
+    else:
+        candidates = iter(
+            _find_candidates_Di(node_order[0], graph_params, state_params, G1_degree)
+        )
```

Comment:
Perhaps we can define `find_candidates` depending on is_directed and then just call it for both cases.
```python
    if not G1.is_directed():
        find_candidates = _find_candidates
    else:
        find_candidates = _find_candidates_Di
    candidates = iter(find_candidates(node_order[0], graph_params, state_params, G1_degree)
    ...  # then later we can just call `find_candidate` and not check which version we are using.
```

Reply:
Yup, seems better